examples/update_flow_control/README.md

```diff
@@ -0,0 +1,108 @@
+method](https://docs.mender.io/development/device-side-api/io.mender.update1#setupdatecontrolmap)
+being used.
+
+## Usage
+
+See [main README](../../README.md) for instructions on how to install this
+package. Once installed, start this example with:
+
+```
+mender-update-flow-control
+```
+
+Which initializes the application, setting the "pause" all Update Control Map,
+and asks the user for further input with the following prompt:
+
+```
+Current map is:
+{
+    "priority": -1,
+    "states": {
+        "ArtifactInstall_Enter": {
+            "action": "pause"
+        },
+        "ArtifactReboot_Enter": {
+            "action": "pause"
+        },
+        "ArtifactCommit_Enter": {
+            "action": "pause"
+        }
+    },
+    "id": "01234567-89ab-cdef-0123-456789abcdef"
+}
+-----------------------
+What do you want to do?
+  0) Pause on all states
+  1) Continue with Installing new software
+  2) Continue with Rebooting
+  3) Continue with Committing new software
+  q) Quit
+
+```
+
+## Implementation
+
+The example application implements two threads that communicate with two queues;
+one thread responsible to manage the Mender DBus API communication, an another
+responsible to deal with the user input:
+
+* `do_handle_update_control_map`: initializes the DBus communication with the
+  Mender client, sets a "pause all" Update Control Map, and enters the main
+  event loop. From this loop two events might happen: that a new map is received
+  in the queue, or that the timeout from the current map is about to expire and
+  the map needs refreshing. On DBus errors, the error message is reported in the
+  corresponding queue and the thread exists.
+
+* `do_main_interactive`: print the selection menu to the user and ask for action
+  to do next. When the user selects an action, the corresponding map is sent to
+  the que for the DBus control thread to handle. It constantly refreshes the
+  terminal to always display the current map on top and the user options menu.
+  If a DBus error is received in the corresponding queue, the error message is
+  printed to the user and the thread exists.
+
+### Note on Python multi-threading
+
+Python (or more specifically, the common implementation CPython) does not
+support concurrent multi-threading. Read more information in [threading module
+documentation](https://docs.python.org/3/library/threading.html) and [Python
+glossary](https://docs.python.org/3/glossary.html#term-global-interpreter-lock).
+If you are planning to use Python in production, consider using [multiprocessing
+module](https://docs.python.org/3/library/multiprocessing.html) instead.
+
+## Troubleshooting
+
+If the application is not working, make sure of the following:
+* `dbus-daemon` is running in your system
+* Mender client is 2.7.0 or newer
+* `DBus` APIs are enabled in the Mender client (they are enabled by default)
+* The `io.mender.UpdateManager.conf` policy file is installed in your system
```

Comment:
And has the right privileges set, no?

Reply:
I will write exactly that, "and has the right privileged", but I honestly don't know which are these.